packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -805,12 +805,14 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         let numBlobs = 0;
         let encodedBlobsSize = 0;
         let decodedBlobsSize = 0;
-        for (const tree of snapshot.trees) {
```

Comment:
It does not feel like it's right change - failures will be further down (in Container / runtime). Even if we such change, don't we want to log an error?

That said, @GaryWilber brought to my attention one such case and I believe it's not valid case, but possibly produced by some of the automation that Gary was experimenting with? If so, I'd not change code for that.

Reply:
Should `IOdspSnapshot` be updated to mark `trees` as able to be `undefined`? Since that's a thing that could happen

Reply:
But we want to know that is it valid or not? Because imo we always expect at least 1 tree.

Reply:
It's a valid response as far as the api itself is concerned. However the majority of fluid runtime scenarios use the create new flow, which makes it so trees is always there.

In my opinion the odsp driver should work for both cases since the api allows both.

If we want to only allow the create new flow to work, the SPO api side should probably be updated to only allow that - this would lead to "trees" being a required property for the response and always contain at least 1 tree.

@marcmasmsft fyi

Reply:
I agree; we can choose to block files that don't start from a snapshot, i.e. no such thing as a file with only ops, but we should then be explicit about the change and block flows that would allow that. Otherwise, trees can be absent from the server response.

---
In reply to: [611856870](https://github.com/microsoft/FluidFramework/pull/5778#discussion_r611856870) [](ancestors = 611856870)

Reply:
I believe that when we supported files created from empty (i.e. no snapshot), then getLatest was returning nothing (I guess failure, though I'm not  sure). I.e. we never took this path, we would get undefined returned from getSnapsthotTree and move to opening socket. If I have to guess, then I'd think something changed over last year in the way this workflow works, but given that we did not hit it (because nobody really opened yearly old files without snapshots), then we never exercised this path.
I'd simplify driver and workflows, not add more paths, We pretty much do not care about files without snapshots, and there is no way today to create file without snapshot (at least in prod scenarios), so we should cut all the other code paths for simplicity,